sound/soc/sof/ipc4-topology.c

```diff
@@ -1035,6 +1036,18 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 			/* at this point amount of dwords */
 			copier_data->gtw_cfg.config_length >>= 2;
 			break;
+		case SOF_DAI_INTEL_ALH:
+		{
+			struct sof_ipc4_alh_configuration_blob *blob;
+
+			blob = devm_kzalloc(sdev->dev, sizeof(*blob), GFP_KERNEL);
+			blob->alh_cfg.count = 1;
+			blob->alh_cfg.mapping[0].alh_id = copier_data->gtw_cfg.node_id;
+			blob->gw_attr.lp_buffer_alloc = 0;
+			copier_data->gtw_cfg.config_length = sizeof(*blob) >> 2;
+			ipc4_copier->copier_config = (uint32_t *)blob;
+			break;
```

Comment:
can we unify all the prepare code for ALH after the call to init_audio_format()?

Reply:
> can we unify all the prepare code for ALH after the call to init_audio_format()?

Yes, done.